src/ORM/DataObject.php

```diff
@@ -2236,7 +2243,14 @@ public function getManyManyComponents($componentName, $id = null)
      */
     public function hasOne()
     {
-        return (array)$this->config()->get('has_one');
+        $hasOne = (array) $this->config()->get('has_one');
+        // Boil down has_one spec to just the class name
+        foreach ($hasOne as $relationName => $spec) {
+            if (is_array($spec)) {
+                $hasOne[$relationName] = DataObject::getSchema()->hasOneComponent($this->objectClass, $relationName);
+            }
+        }
+        return $hasOne;
     }
```

Comment:
This one method is the reason this whole PR works without being a breaking change. Wherever this is called, people can rely on having an associative array of relation name to class name for has_one relations.

Anywhere people are currently using `SomeClass::config()->get('has_one')`, they should probably have been using this instead - and that advice will be included in the changelog.